### ISSUE TITLE
HID PROX SUPPORT

### ISSUE THREAD

**Issue by shayya** [RfidResearchGroup/ChameleonUltra#212]
When will HID PROX LF support please be added? 
I have 6 different lf fobs, and they're all hid prox. You'd think that would be the first lf protocol to be supported

**Comment by GameTec-live** [comment]
as soon as someone finds time and wants to implement it... (this could be you or anyone else)

**Comment by GameTec-live** [comment]
> This seems like a highly demanded feature

Yeah, EVERY somewhat common card that the chameleon could do is in high demand, but people are busy and RRG does god knows what (atleast there arent any recent commits from them). They (Most of the contributors, exception being xianglin1998) are doing that in their freetime and dont get paid...

> As far as I'm aware HID Prox is the the most used LF protocol in the united states

Maybe, but the US is only part of the world XD (So I personallly very rarely encounter LF cards, so i dont care too much about that)

**Comment by shayya** [comment]
Interesting.. where I live about half the systems I encounter are LF (hid prox) and the rest are a mix of mifare and desfire

**Comment by Michael-A-Sloan** [comment]
would love for someone to work on this. I would work on it if I knew how too lol

**Comment by GameTec-live** [comment]
> would love for someone to work on this. I would work on it if I knew how too lol
> 

Get a devkit, HID prox card, A SWD debugger and a pc. Plug them together, set up the nrf build environment and start doing stuff...

**Comment by Michael-A-Sloan** [comment]
> > would love for someone to work on this. I would work on it if I knew how too lol
> > 
> 
> Get a devkit, HID prox card, A SWD debugger and a pc. Plug them together, set up the nrf build environment and start doing stuff...

Since you so knowledgeable what is the stuff part? If I am to spend the kind of money to buy another system I should probably know that part don't you think?

**Comment by GameTec-live** [comment]
"stuff" is whatever you want to work on lol. HID prox id assume as your writing on this issue.
You could also solder the SWD wires to the normal CU, but unlike the devkit it doesnt have headers.

**Comment by TeCHiScy** [comment]
After dig into the code, I've managed to read HID Prox II cards with ChameleonUltra, though more tests need to be done.

**Comment by 4Divinity** [comment]
Can CU read any 125khz cards? Just got mine and I've tried but can't read anything it seems besides the shitty little tags that came with it. Apparently buying a clone from aliexpress might have been a waste

**Comment by tri4ng1e** [comment]
> Can CU read any 125khz cards

CU can read and emulate only one type of LF tag as of now (plain em410x). I hope some day we will be able to read and emulate other types, like HID Prox and others.

**Comment by TeCHiScy** [comment]
> Can CU read any 125khz cards? Just got mine and I've tried but can't read anything it seems besides the shitty little tags that came with it. Apparently buying a clone from aliexpress might have been a waste

yes. different cards just uses different modulations & card formats (decoders). I'm working on HID prox cards, currently on decoders & emulation implementions, as aforesaid.

**Comment by An-git-hub** [comment]
> read HID Prox II cards with ChameleonUltra

Where can I download your dfu-app.zip so that I could upload it to Chameleon Ultra?

**Comment by jamisonderek** [comment]
@TeCHiScy Great Job!!! Do you want any help? You can reach out to me on Discord at @CodeAllNight (note: I'm MrDerekJamison on YouTube and JamisonDerek on GitHub).  I've added LF protocols to Chameleon Ultra (on the firmware/CLI side) so I know that part of the puzzle. I haven't looked at the supporting apps yet.

I'd love to see how you are doing the FSK demodulation (I checked your fork but didn't see any commits). I just finished implementing [Viking protocol](https://github.com/jamisonderek/ChameleonUltra/tree/jamisonderek/viking) which is using Manchester, RF/32 but I don't see how to do FSK2a modulation since `register_rio_callback(GPIO_INT0_callback);` only gives falling edge & we need to know every 8 or 10 cycles to demod.

I can see you are clearly getting Wiegand bits, facility and card number -- so you obviously figured it out. I've made [videos](https://www.youtube.com/playlist?list=PLM1cyTMe-PYKJeHWkN8vKntEyqmqlj_kY) on various LF protocols, but I need help understanding how to get the raw data for FSK2a (I understand how to demod if I know a bunch of FC/8, FC/10 values).

**Comment by TeCHiScy** [comment]
> [@TeCHiScy](https://github.com/TeCHiScy) Great Job!!! Do you want any help? You can reach out to me on Discord at [@codeallnight](https://github.com/codeallnight) (note: I'm MrDerekJamison on YouTube and JamisonDerek on GitHub). I've added LF protocols to Chameleon Ultra (on the firmware/CLI side) so I know that part of the puzzle. I haven't looked at the supporting apps yet.
> 
> I'd love to see how you are doing the FSK demodulation (I checked your fork but didn't see any commits). I just finished implementing [Viking protocol](https://github.com/jamisonderek/ChameleonUltra/tree/jamisonderek/viking) which is using Manchester, RF/32 but I don't see how to do FSK2a modulation since `register_rio_callback(GPIO_INT0_callback);` only gives falling edge & we need to know every 8 or 10 cycles to demod.
> 
> I can see you are clearly getting Wiegand bits, facility and card number -- so you obviously figured it out. I've made [videos](https://www.youtube.com/playlist?list=PLM1cyTMe-PYKJeHWkN8vKntEyqmqlj_kY) on various LF protocols, but I need help understanding how to get the raw data for FSK2a (I understand how to demod if I know a bunch of FC/8, FC/10 values).

Hey, I would reach you out on Discord some time later.
